homework/index.js

```diff
@@ -30,18 +15,178 @@
     return elem;
   }
 
-  function main(url) {
-    fetchJSON(url, (err, data) => {
-      const root = document.getElementById('root');
-      if (err) {
-        createAndAppend('div', root, { text: err.message, class: 'alert-error' });
+  function createHeader(root) {
+    const header = createAndAppend('header', root, {
+      class: 'header',
+    });
+    createAndAppend('p', header, {
+      text: 'HYF Repositories',
+    });
+    return header;
+  }
+
+  function createOptions(repositoryList, header) {
+    const select = createAndAppend('select', header, {
+      id: 'selector',
+    });
+    for (let i = 0; i < repositoryList.length; i++) {
+      const repository = repositoryList[i];
+      createAndAppend('option', select, {
+        text: repository.name,
+        value: i,
+      });
+    }
+  }
+
+  function createMainContainer(root) {
+    const mainContainer = createAndAppend('div', root, {
+      id: 'container',
+    });
+    return mainContainer;
+  }
+
+  function containerListContributor(mainContainer) {
+    const leftContainer = createAndAppend('div', mainContainer, {
+      class: 'left-block frame',
+    });
+    const table = createAndAppend('table', leftContainer);
+    const listContributor = createAndAppend('tbody', table);
+    return listContributor;
+  }
+
+  function createContributorContainer(mainContainer) {
+    const rightContainer = createAndAppend('div', mainContainer, {
+      class: 'right-block frame',
+    });
+    createAndAppend('p', rightContainer, {
+      text: 'Contributions',
+      class: 'contributor',
+    });
+
+    return createAndAppend('ul', rightContainer, {
+      class: 'list1',
+    });
+  }
+
+  function renderContributors(contributorList, contributors) {
+    contributorList.innerHTML = '';
+
+    for (let i = 0; i < contributors.length; i++) {
+      const li = createAndAppend('li', contributorList, {
+        class: 'contributor-item',
+      });
+      const linkFor = createAndAppend('a', li, {
+        target: '_blank',
+        href: contributors[i].html_url,
+      });
+      createAndAppend('img', linkFor, {
+        src: contributors[i].avatar_url,
+        class: 'avatar',
+        height: 48,
+      });
+      const divCont = createAndAppend('div', linkFor, {
+        class: 'contributor-data',
+      });
+      createAndAppend('div', divCont, {
+        text: contributors[i].login,
+      });
+      createAndAppend('div', divCont, {
+        text: contributors[i].contributions,
+        class: 'badge',
+      });
+    }
+  }
+
+  async function renderRep(listContributors, repo, root, contributorList) {
```

Comment:
Avoid unnecessary abbreviations. What do you mean with `Rep`? If you can type the full word `Contributors` in line 71, why then shouldn't you type the full word `Repository` here?

Reply:
Definitely. .